meta/Meta.cpp

```diff
@@ -6937,6 +6937,124 @@ void Meta::meta_sai_on_bfd_session_state_change(
     }
 }
 
+void Meta::meta_sai_on_ha_set_event_single(
+    _In_ const sai_ha_set_event_data_t& data)
+{
+    SWSS_LOG_ENTER();
+
+    auto ot = objectTypeQuery(data.ha_set_id);
+
+    bool valid = false;
+
+    switch (ot)
+    {
+        // TODO hardcoded types, must advance SAI repository commit to get metadata for this
```

Comment:
you already have this commit, metadata can be used to support that, check other notifications implementation in this file how to do this

Reply:
Hi @kcudnik - I checked the latest libsaimetadata on the current head, but didn't see the metadata related to HA notifications? Is it expected?

Reply:
i was thinking about similar to this: https://github.com/sonic-net/sonic-sairedis/blob/master/meta/Meta.cpp#L6768
please follow https://github.com/sonic-net/sonic-sairedis/blob/master/meta/Meta.cpp#L7232

Reply:
Hi @kcudnik - yes I did with commit https://github.com/sonic-net/sonic-sairedis/pull/1583/commits/7dbe8e589034d749283a33bdd0eea1fe04969d08

But the metadata is missing, hence the build couldn't pass.

Reply:
Hi @kcudnik - even if I revert the commit: 
![image](https://github.com/user-attachments/assets/966011cd-1d33-4dcb-8e6c-97b00b462e96)

The ha types are in the extension headers, is that why? 
https://github.com/opencomputeproject/SAI/blob/2f583d4c5477db13a3680cf2e8e622971d8f4fe2/experimental/saitypesextensions.h#L73

Reply:
You need to cast to int before switch

Reply:
Hi @kcudnik  - done already. Please check on the latest commit.